interfaces/saml/v0/README.md

```diff
@@ -0,0 +1,75 @@
+# `saml`
+
+## Overview
+
+This relation interface describes the expected behavior between of any charm claiming to be able to interface with a SAML Provider and the SAML Provider itself. This interface has been defined to support charms that require Launchpad group information. Other Identity Providers can be used interchangeably as long as custom attributes are not required.
+
+## Usage
+
+In most cases, this will be accomplished using the [saml library](https://github.com/canonical/saml-integrator-operator/blob/main/lib/charms/saml_integrator/v0/saml.py), although charm developers are free to provide alternative libraries as long as they fulfil the behavioural and schematic requirements described in this document.
+
+## Direction
+
+The `saml` interface implements a provider pattern.
+The requirer is a charm that wishes to act as a SAML Service Provider, and the provider is a charm exposing a SAML Identity Provider.
+
+```mermaid
+flowchart TD
+    Provider -- metadata_url, entity_id, single_sign_on_service_redirect_url, single_sign_on_service_redirect_binding, x509certs --> Requirer
+```
+
+## Behavior
+
+The requirer and the provider must adhere to a certain set of criteria to be considered compatible with the interface.
+
+### Provider
+
+- Is expected to provide the metadata_url in the relation databag.
+- Is expected to provide the entity_id in the relation databag.
+- Is expected to provide the single_sign_on_service_redirect_url in the relation databag.
+- Is expected to provide the single_sign_on_service_redirect_binding in the relation databag.
+- Is expected to provide the x509certs in the relation databag.
+
+### Requirer
+
+- Is not expected to provide anything
+
+## Relation Data
+
+### Provider
+
+[\[JSON Schema\]](./schemas/provider.json)
+
+Provider provides the SAML configuration. It should be placed in the **application** databag.
+
+#### Example
+
+```yaml
+related-units: {}
+application_data: {
+  "metadata_url": "https://login.ubuntu.com/saml/metadata",
+  "entity_id": "https://login.ubuntu.com",
+  "single_sign_on_service_redirect_url": "https://login.ubuntu.com/saml/",
+  "single_sign_on_service_redirect_binding": "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect",
+  "single_logout_service_redirect_url": "https://login.ubuntu.com/+logout",
```

Comment:
For the redirect url and binding the prefix is `single_sign_on` and here it's `single`, we should use the same for both

Reply:
The prefixes are "single_singn_on" and "single_logout". These have already been agreed as part of a spec